pkg/controller/volume/attach_detach_controller.go

```diff
@@ -414,14 +415,14 @@ func (adc *attachDetachController) getPVSpecFromCache(
 	name string,
 	pvcReadOnly bool,
 	expectedClaimUID types.UID) (*volume.Spec, error) {
-	pvObj, exists, err := adc.pvInformer.GetStore().Get(name)
+	pvObj, exists, err := adc.pvInformer.GetStore().GetByKey(name)
 	if pvObj == nil || !exists || err != nil {
 		return nil, fmt.Errorf(
 			"failed to find PV %q in PVInformer cache. %v", name, err)
 	}
 
 	pv, ok := pvObj.(*api.PersistentVolume)
-	if ok || pv == nil {
```

Comment:
Also, if this code got in, it means that there's a hole in the unit coverage for this method